OpenSim/Common/Test/testTRCFileAdapter.cpp

```diff
@@ -60,77 +40,130 @@ void testFailed(const std::string& filename,
 void compareFiles(const std::string& filenameA, 
                   const std::string& filenameB) {
     // Delimiters include carriage return and newline.
-    const std::string delims{" \t\r\n"};
+    const std::string delims{"\t\r\n"};
 
     std::ifstream fileA{filenameA};
     std::ifstream fileB{filenameB};
-    while(fileA && fileB) {
-        const auto tokenA = getNextToken(fileA, delims);
-        const auto tokenB = getNextToken(fileB, delims);
-        if(tokenA != tokenB) {
-            double d_tokenA{};
-            double d_tokenB{};
-            try {
-                d_tokenA = std::stod(tokenA);
-                d_tokenB = std::stod(tokenB);
-            } catch(std::invalid_argument&) {
-                testFailed(filenameA, tokenA, tokenB);
-            }
 
-            if(d_tokenA != d_tokenB && 
-               !(std::isnan(d_tokenA) && 
-                 std::isnan(d_tokenB))) {
-                testFailed(filenameA, tokenA, tokenB);
+    // Use a STOFileAdapter just to access the common
+    // FileAdapter utilities.
+    OpenSim::STOFileAdapter sfa;
```

Comment:
By using `FileAdapter` utilities to to test the file adapters, is it possible to have a bug in `getNextLine()` and the tests would still pass?